spanner/transaction.go

```diff
@@ -1049,6 +1050,53 @@ func (t *ReadWriteTransaction) runInTransaction(ctx context.Context, f func(cont
 	return ts, nil
 }
 
+// ReadWriteTransactionStmtBased provides a wrapper of ReadWriteTransaction in
+// order to run a read-write transaction in a statement-based way.
+//
+// This struct is returned by client.BeginReadWriteTransaction and contains
+// Commit() and Rollback() methods to end a transaction.
+type ReadWriteTransactionStmtBased struct {
```

Comment:
Thanks for doing this :-)

Reply:
I rename it to `ReadWriteStmtBasedTransaction` which sounds better.